File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.js
const fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input.shift());
let word = input.map((v)=>v.trim())
let first = word.shift().split('');

for (let i=0; i<word.length; i++) {
  for (let j=0; j<word[i].length; j++) {
    if (word[i][j] === first[j]) {
      continue;
    } else {
      first[j] = '?'
    }
  }
}

console.log(first.join(''));

File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.py
import sys
input = sys.stdin.readline

# ?에는 아무 문자가 와도 된다. 
# 가능하면 ?를 적게 써야하는 패턴을 return 해야 한다. 

n = int(input().rstrip())
words = [input().rstrip() for _ in range(n)]
f_word = list(words[0])
answer = ''

for i in range(1,n):
    for j in range(len(words[i])):
        if words[i][j] == f_word[j]:
            continue
        else:
            f_word[j] = '?'

print(''.join(f_word))